db/osqlblockproc.c

```diff
@@ -608,6 +642,81 @@ const char *osql_reqtype_str(int type)
     return typestr[type];
 }
 
+void setup_reorder_key(int type, osql_sess_t *sess, struct ireq *iq, char *rpl,
+                       oplog_key_t *key)
+{
+    key->tbl_idx = USHRT_MAX;
+    if (type == OSQL_USEDB) {
+        const char *get_tablename_from_rpl(const char *rpl);
+        const char *tablename = get_tablename_from_rpl(rpl);
+        assert(tablename); //table or queue name
+        if (tablename && !is_tablename_queue(tablename, strlen(tablename))) {
+            struct dbtable *tbl = get_dbtable_by_name(tablename);
```

Comment:
Cant do this without a lock. Caching tbl that may be dropped will result in unexpected behavior. 
Cache the tablename instead and get the dbtable by name every time you need to create a new genid, guarded by appropriate lock. Tbl index can be crc32 of tablename.